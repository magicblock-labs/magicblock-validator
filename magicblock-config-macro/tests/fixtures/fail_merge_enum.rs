use magicblock_config_macro::Mergeable;

#[derive(Default, Mergeable)]
enum TestEnum {
    A(u32),
    B(String),
}

fn main() {}
