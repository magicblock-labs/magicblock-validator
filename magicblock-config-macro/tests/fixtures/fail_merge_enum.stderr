error: no default declared
 --> tests/fixtures/fail_merge_enum.rs:3:10
  |
3 | #[derive(Default, Mergeable)]
  |          ^^^^^^^
  |
  = help: make a unit variant default by placing `#[default]` above it
  = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Merge can only be derived for structs
 --> tests/fixtures/fail_merge_enum.rs:4:1
  |
4 | / enum TestEnum {
5 | |     A(u32),
6 | |     B(String),
7 | | }
  | |_^
